Jenkinsfile

```diff
@@ -0,0 +1,173 @@
+pipeline {
+    agent { label 'upbound-gce' }
+
+    parameters {
+        booleanParam(defaultValue: true, description: 'Execute pipeline?', name: 'shouldBuild')
+    }
+
+    options {
+        disableConcurrentBuilds()
+        timestamps()
+    }
+
+    environment {
+        RUNNING_IN_CI = 'true'
+        REPOSITORY_NAME = "${env.GIT_URL.tokenize('/')[3].split('\\.')[0]}"
+        REPOSITORY_OWNER = "${env.GIT_URL.tokenize('/')[2]}"
+        DOCKER = credentials('dockerhub-upboundci')
+        AWS = credentials('aws-upbound-bot')
+        GITHUB_UPBOUND_BOT = credentials('github-upbound-jenkins')
```

Comment:
This uses an Upbound Bot, so this will likely break on recording test coverage, etc.